tortoise/src/main/scala/Compiler.scala

```diff
@@ -164,9 +164,9 @@ object Compiler extends CompilerLike {
     else
       defs.head
     if (commands)
-      handlers.commands(pd.statements, true, !raw)
+      handlers.commands(pd.statements, true, !raw)(compilerFlags = compilerFlags, compilerContext = new CompilerContext(blockLevel = 0, source = wrapped))
     else
-      handlers.reporter(pd.statements.stmts(1).args(0))
+      handlers.reporter(pd.statements.stmts(1).args(0))(compilerFlags = compilerFlags, compilerContext = new CompilerContext(blockLevel = 0, source = wrapped))
```

Comment:
We can retain the old non-verbosity of the "implicit" machinery here by providing an implicit function that builds a `CompilerContext` from the source code.  But, since the source is a `String`, we want to wrap it in a simple wrapper class, to avoid accidentally triggering the implicit conversion unintentionally.

An example solution of how to accomplish this can be found [here](https://gist.github.com/TheBizzle/ff52c182baa60e89fe5c81e62cf82460).